### PR TITLE
test: Condition tests for web-crypto-backend

### PR DESCRIPTION
**Pull request by seebees**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

# Check any applicable:
- [ ] Were any files moved? Moving files changes their URL, which breaks all hyperlinks to the files.

### CODE REVIEW

File: modules/web-crypto-backend/test/backend-factory.test.ts [hunk 1]
```diff
@@ -189,3 +195,25 @@ describe('webCryptoBackendFactory', () => {
...
 })
+
+describe('getNonZeroByteBackend', () => {
+  it('gets a subtle backend', () => {
+    const test = getNonZeroByteBackend(fixtures.subtleFallbackSupportsZeroByteGCM)
+    expect(test === fixtures.subtleFallbackSupportsZeroByteGCM.subtle).to.equal(true)
+  })
+
+  it('Precondition: A backend must be passed to get a non zero byte backend.', () => {
+    expect(() => getNonZeroByteBackend(false)).to.throw('No supported backend.')
```
Comment (lizroth) [inline modules/web-crypto-backend/test/backend-factory.test.ts]: Is there a convenient way to make the meaning of the `false` more intuitive to the reader (e.g. by naming it, or adding a comment for what it does)?

File: modules/web-crypto-backend/test/backend-factory.test.ts [hunk 2]
```diff
@@ -189,3 +195,25 @@ describe('webCryptoBackendFactory', () => {
...
+})
+
+describe('getZeroByteSubtle', () => {
+  it('gets a subtle backend', () => {
+    const test = getZeroByteSubtle(fixtures.subtleFallbackSupportsZeroByteGCM)
+    expect(test === fixtures.subtleFallbackSupportsZeroByteGCM.subtle).to.equal(true)
+  })
+
+  it('Precondition: A backend must be passed to get a zero byte backend.', () => {
+    expect(() => getZeroByteSubtle(false)).to.throw('No supported backend.')
```
Comment (lizroth) [inline modules/web-crypto-backend/test/backend-factory.test.ts]: Ditto

File: modules/web-crypto-backend/test/backend-factory.test.ts (hunk 1 above)
Comment (seebees) [reply modules/web-crypto-backend/test/backend-factory.test.ts]: Hum... These functions operate on backends returned.  I can add a comment.  But the idea is that if a backend does not exist, I return false to be explicit.  These helper functions are just handling that...

Comment (lizroth) [reply modules/web-crypto-backend/test/backend-factory.test.ts]: We chatted about this offline and I can't think of a non-confusing way to express what I want here, so I would say let's just leave it.

**Pull request merged**